server/src/main/java/org/opensearch/index/query/BoolQueryBuilder.java

```diff
@@ -402,8 +400,10 @@ protected QueryBuilder doRewrite(QueryRewriteContext queryRewriteContext) throws
             return any.get();
         }
 
-        changed |= rewriteMustNotRangeClausesToShould(newBuilder, queryRewriteContext);
-        changed |= rewriteMustClausesToFilter(newBuilder, queryRewriteContext);
+        // Note: must-not-to-should optimization is now handled by MustNotToShouldRewriter in the query rewriting infrastructure
```

Comment:
we can just remove these lines instead of commenting them

Reply:
Resolved